packages/web-components/src/button/button.ts

```diff
@@ -254,20 +294,54 @@ export class BaseButton extends FASTElement {
       return;
     }
 
+    if (this.popoverTargetElement) {
+      switch (this.popovertargetaction) {
+        case 'hide':
+          this.popoverTargetElement?.hidePopover();
+          break;
+        case 'show':
+          this.popoverTargetElement?.showPopover();
+          break;
+        case 'toggle':
+        default:
+          this.popoverTargetElement?.togglePopover(!this.popoverOpen);
+          break;
+      }
+    }
+
     this.press();
     return true;
   }
 
-  connectedCallback(): void {
+  public connectedCallback(): void {
     super.connectedCallback();
     this.elementInternals.ariaDisabled = `${!!this.disabledFocusable}`;
+
+    if (this.popovertarget) {
+      this.setPopoverTargetElement(this.popovertarget);
+    }
   }
 
   constructor() {
     super();
     this.elementInternals.role = 'button';
   }
 
+  /**
+   * Sets up handling for the popover target element
+   * @param element - The element to reference
+   */
+  private setPopoverTargetElement(element: string | undefined) {
+    this.popoverTargetElement = element ? document.getElementById(element) : undefined;
+
+    this.popoverTargetElement?.addEventListener(
+      'toggle',
+      e =>
+        (this.popoverOpen =
+          (e as Event & { newState: 'open' | 'closed'; oldState: 'open' | 'closed' }).newState === 'open'),
```

Comment:
I don't love this but in the case of toggling the popover, it seems like we need to force it closed if it is open.

Reply:
Yeah I got the same issue with Dropdown, if the invoker element is clicked, first the popover element would close since a click event happened outside of it, then the click event fires on the invoker, which is too late to prevent any default behavior. +1 to this solution.